unity/EditorPlugin/AssetPostprocessors/CsprojAssetPostprocessor.cs

```diff
@@ -353,32 +369,39 @@ private static bool ApplyManualCompilerSettingsReferences(XElement projectConten
 
       foreach (var referenceName in referenceList)
       {
-        string hintPath = null;
-
         var name = referenceName;
         if (new FileInfo(name).Extension != ".dll")
           name += ".dll"; // RIDER-15093
+        
+        var hintPath = GetHintPath(name);
+        ApplyCustomReference(Path.GetFileName(hintPath), projectContentElement, xmlns, hintPath);
+      }
 
-        if (PluginSettings.SystemInfoRiderPlugin.operatingSystemFamily == OperatingSystemFamilyRider.Windows)
-        {
-          var unityAppBaseFolder = Path.GetDirectoryName(EditorApplication.applicationPath);
-          var monoDir = new DirectoryInfo(Path.Combine(unityAppBaseFolder, "MonoBleedingEdge/lib/mono"));
-          if (!monoDir.Exists)
-            monoDir = new DirectoryInfo(Path.Combine(unityAppBaseFolder, "Data/MonoBleedingEdge/lib/mono"));
+      return true;
+    }
 
-          var newestApiDir = monoDir.GetDirectories("4.*").LastOrDefault();
-          if (newestApiDir != null)
-          {
-            var dllPath = new FileInfo(Path.Combine(newestApiDir.FullName, name));
-            if (dllPath.Exists)
-              hintPath = dllPath.FullName;
-          }
-        }
+    [CanBeNull]
+    private static string GetHintPath(string name)
+    {
+      string hintPath = null;
 
-        ApplyCustomReference(name, projectContentElement, xmlns, hintPath);
+      if (PluginSettings.SystemInfoRiderPlugin.operatingSystemFamily == OperatingSystemFamilyRider.Windows)
```

Comment:
Why is this only for Windows? Why don't we do the same for Mac/Linux? It works without hint path on Mac, so what's the issue we're solving for Windows?

Reply:
On Windows MSBuild will 100% resolve it from dotnet framework targets path 
On Mac/Linux in case when xbuild from Unity is used - it will resolve it from Unity reference assemblies.
However you are right about the case, when non-Unity MsBuild is used. I will add hintpaths for all os.